### PR TITLE
mega menu

### PR DESCRIPTION
**Pull request by korgan00**
## Changes

Fix https://github.com/Qiskit/web-components/issues/74
Fix https://github.com/Qiskit/qiskit.org/issues/2100
Creates mega-menu web-component version

## Implementation details

Enhances the previous version with the new interface and fixes the [bug issue](https://github.com/Qiskit/qiskit.org/issues/2100) where the user cannot close the mega menu.

Code Changes:

- The content structure is slightly different. It adds a new level called `MegaMenuDropdownBlock`. It doesn't support columns because it will do it automatically.
``` Typescript
interface MegaMenuDropdownGroup {
  title: NavLink;
  content: NavLink[];
}

interface MegaMenuDropdownBlock {
  title: string;
  content: MegaMenuDropdownGroup[];
}

export type MegaMenuDropdownContent = MegaMenuDropdownBlock[];
```
- The segmentation should be treated using the event in the component.

### CODE REVIEW

File: components/mega-menu-dropdown/index.scss [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+      column-count: 3;
+      column-count: min(var(--elements-count, 3), 3);
+    }
+  }
+
+  &__group {
+    display: flex;
+    flex-direction: column;
+    break-inside: avoid;
```
Comment (techtolentino) [inline components/mega-menu-dropdown/index.scss:117]: Small comment: I think that between content groups, we should have 48px spacing between. Showing in the video capture below 👇 

https://user-images.githubusercontent.com/6276074/169888461-7ea2c62a-6e25-43bf-bf28-030d8aeb7718.mov

File: components/mega-menu-dropdown/index.scss [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  width: 12 * 4rem;
+  height: 32rem;
+  padding: 0 $spacing-05;
+  background-color: var(--qiskit-color-background-primary);
+  box-shadow: 0 4px 4px rgb(0 0 0 / 25%);
+
+  &__block {
+    display: flex;
+    flex-direction: column;
+    padding-bottom: 1rem;
```
Comment (techtolentino) [inline components/mega-menu-dropdown/index.scss:88]: This can use the Carbon spacing

**Review by techtolentino — CHANGES_REQUESTED** [review]
Small suggestions @korgan00 -- this looks amazing 👏

**Review by techtolentino — DISMISSED** [review]
This works great, @korgan00 

- pulled down the branch
- explored the component (👏 )
- had offline discussions about any questions I had

thank you for answering my questions, too.

LGTM

File: components/mega-menu-dropdown/index.ts [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+
+@customElement('mega-menu-dropdown')
+export class MegaMenuDropdown extends LitElement {
+  static styles = [styles];
+
+  @query('.filter__input')
+  protected _filterInput!: HTMLInputElement;
+
+  @query('.app-mega-dropdown')
+  protected _mainElement!: HTMLInputElement;
```
Comment (abdonrd) [inline components/mega-menu-dropdown/index.ts:47]:
```suggestion
  protected _mainElement!: HTMLArticleElement;
```

**Review by abdonrd — CHANGES_REQUESTED** [review]

**Comment by abdonrd** [comment]
Right now each link has this `padding-bottom: 1rem;` which makes the yellow section clickable:

Screen Shot 2022-05-24 at 16 37 39

What do you think about to center the text to have same clickable top/bottom?

File: components/mega-menu-dropdown/index.ts (hunk 3 above)
Comment (korgan00) [reply components/mega-menu-dropdown/index.ts]: I used `HTMLElement` because `Article` does not exist.

**Review by abdonrd — APPROVED** [review]

**Pull request merged**